sql/src/main/java/io/crate/executor/transport/AlterTableOperation.java

```diff
@@ -107,7 +108,58 @@ public AlterTableOperation(ClusterService clusterService,
         return result;
     }
 
-    private class ResultSetReceiver implements ResultReceiver {
+    public CompletableFuture<Long> executeAlterTableOpenClose(final AlterTableOpenCloseAnalyzedStatement analysis) {
+        List<CompletableFuture<Long>> results = new ArrayList<>(2);
```

Comment:
In the case of an open partitioned table with a closed partition, how would we handle this? Since the table is already "open", yet opening it would open all closed partitions underneath it.